tripy/examples/segment-anything-model-v2/video_demo.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Not a contribution
+# Changes made by NVIDIA CORPORATION & AFFILIATES enabling SAM2 with Tripy or otherwise documented as
+# NVIDIA-proprietary are not a contribution and subject to the following terms and conditions:
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from sam2.build_sam import build_sam2_video_predictor
+import numpy as np
+import matplotlib.pyplot as plt
+from PIL import Image
+from demo_utils import process_and_show_mask as show_mask
+from typing import Optional
+
+import os
+import time
+
+
+def compute_mask_properties(mask):
+    # Ensure we have a boolean array
+    test_mask = np.asarray(mask, dtype=bool)
+
+    # Calculate basic stats
+    volume = np.sum(test_mask)
+
+    # Calculate centroid (center of mass)
+    indices = np.where(test_mask)
+    assert volume > 0
+    centroid = tuple(float(np.mean(idx)) for idx in indices)
+    return volume, centroid
+
+
+def main(video_dir: str, save_path: Optional[str] = None):
+    """
+    Main execution function.
+
+    Args:
+        video_path (str): Path to where video frames are stored
+        save_path (str, optional): Directory to save visualizations
+
+    Returns:
+        Dict[str, np.ndarray]: Processing results
+    """
+
+    sam2_checkpoint = "./checkpoints/sam2.1_hiera_large.pt"
```

Comment:
Maybe this should be part of the config?

Reply:
Let's keep it this way to maintain parity with OSS SAM2.